openhtf/io/output/__init__.py

```diff
@@ -12,15 +12,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Openhtf output modules."""
+"""This module contains support for various built-in output mechanisms.
 
-import openhtf.io.output.json_factory
-import openhtf.io.output.mfg_inspector
+Here, a base OutputToFile class is implemented to provide simple output to
+a file via the pickle serialization mechanism. It can be subclassed to implement
+alternative serialization schemes, see json_factory.py and mfg_inspector.py for
+examples.
+"""
 
-# pylint: disable=invalid-name
+import base64
+import cPickle as pickle
+import os
+import tempfile
+from openhtf.util import data
 
-# Classes used for output
-OutputToJSON = json_factory.OutputToJSON
-UploadToMfgInspector = mfg_inspector.UploadToMfgInspector
-OutputToTestRunProto = mfg_inspector.OutputToTestRunProto
-UploadOrOutput = mfg_inspector.UploadOrOutput
+
+class OutputToFile(object):
+  """Class for formatting filename and writing to file."""
+
+  def __init__(self, filename_pattern=None):
```

Comment:
Hm, I think the default value of None was to use OutputToJSON as a JSONEncoder without doing the actual file output, but that doesn't really make sense in this context, so I think it's safe to require the filename_pattern.